docker/pravega/Dockerfile

```diff
@@ -9,6 +9,9 @@
 #
 FROM java:8
 
+RUN apt-get update && apt-get install -y -q rpcbind nfs-common \
```

Comment:
> I think I share the same concern as @tkaitchuck. Why should we do apt-get and apt-install here?

This is needed so that NFS shares can be mounted inside of the container. The Dockerfile entrypoint (`docker/pravega/entrypoint.sh`) does that when `TIER2_STORAGE=FS`, `MOUNT_IN_CONTAINER=true` and `NFS_MOUNT` is set.


> If we do this, are we forcing everyone using Pravega with docker to install these packages?

Anyone using Pravega with Docker would likely be using the published image. They won't have to install the packages themselves, but yes, the published image will include these packages even if they don't use the `MOUNT_IN_CONTAINER` option.


> Is there any potential conflict?

No. These packages are only going to be installed in the Docker image and will not conflict with the host system. This is done at build/publish time.


> Is there a way of doing it via the docker compose files so that we do it only for the simple deployments we intend it for?

This would make it less useful. The published Docker image can be used in many other ways than docker compose.


> What if it is not ubuntu?

This will always be ubuntu as the image starts off with an ubuntu image (`FROM java:8`). If someone was running pravega on their own, they would be responsible to manage the NFS share mount points on their own.